pkg/common/common_test.go

```diff
@@ -0,0 +1,62 @@
+package common
+
+import (
+	"reflect"
+	"testing"
+)
+
+func Test_parseDynatraceConfigFile(t *testing.T) {
+	tests := []struct {
+		name       string
+		yamlString string
+		want       DynatraceConfigFile
+		wantErr    bool
+	}{
+		{
+			name:       "empty string",
+			yamlString: "",
+			want:       DynatraceConfigFile{},
+			wantErr:    false,
+		},
+		{
+			name: "valid yaml no dashboard",
+			yamlString: `spec_version: '0.1.0'
+dtCreds: dyna`,
```

Comment:
Any possibility to improve formatting ? May be parse yaml from some resource files ?

Reply:
Nice suggestion! I found this can be done using the embed package, e.g. to initialize `testYAML` with the contents of `test.yaml`:

```
import _ "embed"

//go:embed test.yaml
var testYAML string
```
My question: is this more or less fragile than embedding snippets directly in the struct?

Reply:
This is a matter of taste. I also like it if the unit test is fully self-contained.